server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -378,6 +382,37 @@ public static IndexMergePolicy fromString(String text) {
          */
         new ByteSizeValue(Translog.DEFAULT_HEADER_SIZE_IN_BYTES + 1, ByteSizeUnit.BYTES),
         new ByteSizeValue(Long.MAX_VALUE, ByteSizeUnit.BYTES),
+        new Setting.Validator<ByteSizeValue>() {
+            @Override
+            public void validate(final ByteSizeValue value) {}
+
+            @Override
+            public void validate(ByteSizeValue value, Map<Setting<?>, Object> settings) {
+                boolean isCompositeIndex = (boolean) settings.get(StarTreeIndexSettings.IS_COMPOSITE_INDEX_SETTING);
+                ByteSizeValue compositeIndexMaxFlushThreshold = (ByteSizeValue) settings.get(
+                    COMPOSITE_INDEX_MAX_TRANSLOG_FLUSH_THRESHOLD_SIZE_SETTING
+                );
+                if (isCompositeIndex && value.compareTo(compositeIndexMaxFlushThreshold) > 0) {
+                    throw new IllegalArgumentException(
+                        String.format(
+                            Locale.ROOT,
+                            "You can configure '%s' with upto '%s' for composite index",
+                            INDEX_TRANSLOG_FLUSH_THRESHOLD_SIZE_SETTING.getKey(),
+                            compositeIndexMaxFlushThreshold
+                        )
+                    );
+                }
+            }
+
+            @Override
+            public Iterator<Setting<?>> settings() {
+                final List<Setting<?>> settings = Arrays.asList(
+                    StarTreeIndexSettings.IS_COMPOSITE_INDEX_SETTING,
+                    COMPOSITE_INDEX_MAX_TRANSLOG_FLUSH_THRESHOLD_SIZE_SETTING
```

Comment:
Declaring a cluster setting here doesn't help I guess

Reply:
It does help fetching the value, it works fine - since its final as well, we can use it

Reply:
but let me check node settings to see if we can make the dependent setting dynamic

Reply:
All it does is load the default value which is in the code, I don't think the yml override even gets loaded. 

I'm thinking why not make this an index setting rather, because each index will have varied cardinality and default max can be 512 mb which user can override if needed.

Reply:
Made this change similar to `validateRefreshIntervalSettings` , thanks for the find @mgodwan